src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,194 @@
+package duke.util;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Manages persistent storage of information.
+ */
+public class Storage {
+    private final String path;
+
+    /**
+     * Constructor for Storage.
+     * @param path Path indicating where information should be stored or retrieved from.
+     */
+    public Storage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Reads from a saved file. If a file does not yet exist, it creates a new file.
+     * @return ArrayList of Tasks created from saved tasks.
+     * @throws DukeException Occurs whenever a file could not be found or created.
+     */
+    public ArrayList<Task> readFile() throws DukeException {
+        File file = new File(path);
+        ArrayList<Task> taskList = new ArrayList<>();
+        int lineIndex = 1;
+
+        if(!file.exists()) {
+            try {
+                createFile(file);
+            } catch (DukeException e) {
+                throw new DukeException("Could not create a new file.");
+            }
+        }
+
+        try {
+            Scanner scanner = new Scanner(file);
```

Comment:
Maybe you could consider closing your scanner after you are done reading the file?